skymp5-server/cpp/server_guest_lib/ActionListener.cpp

```diff
@@ -480,6 +480,20 @@ void ActionListener::OnChangeValues(const RawMessageData& rawMsgData,
     stamina = CropStaminaRegeneration(stamina, timeAfterRegeneration, actor);
   }
 
+  if (health != healthPercentage || magicka != magickaPercentage ||
```

Comment:
Please perform float comparison with epsilons: https://stackoverflow.com/questions/4915462/how-should-i-do-floating-point-comparison
(we should treat 0.5 as equal to 0.500001 for example)